File: backend/app/schemas/github.py
# backend/app/schemas/github.py
from pydantic import BaseModel

class GithubUser(BaseModel):
    id: int
    login: str
    avatar_url: str


class GithubUserDetails(BaseModel):
    id: int
    login: str
    avatar_url: str
    html_url: str
    bio: str | None = None
    location: str | None = None
    created_at: str
    followers: int
    following: int
    public_repos: int

class LanguagesStat(BaseModel):
    name: str
    bytes: int
    percentage: float

File: frontend/gittalent-radar/src/types/github.ts
export type GithubUser = {
  id: number;
  login: string;

  avatar_url: string;
  html_url: string;
  bio: string | null;
  location: string | null;

  created_at: string;

  followers: number;
  following: number;

  public_repos: number;
};

export type LanguageStat = {
  name: string;
  bytes: number;
  percentage: number;
};

export interface FeaturedUser {
  id: number;
  login: string;
  avatar_url: string;
}